script/model.py

```diff
@@ -4,141 +4,350 @@
 import numpy as np
```

Comment:
Consider adding a file/top-level docstring to describe the purpose of this file.

Reply:
added to all files